packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1446,7 +1452,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     private formMetadata(): IContainerRuntimeMetadata {
         return {
             ...this.createContainerMetadata,
-            summaryCount: this.summaryCount,
+            // Increment the summary number for the next summary that will be generated,
+            summaryNumber: this.nextSummaryNumber++,
```

Comment:
Does it make sense to have this `formMetadata()` method increment the `this.nextSummaryNumber`? This method looks to be called at the end of the summarize workflow, but will this always be the case?

Reply:
That's a good point. I initially had it in `submitSummary` function after summarize is called. The reason I moved it here is because the expectation is that this will be called on every summary to add the metadata. Also, the first summary (via `createSummary()`) goes down a different path.
You are right that if `formMetadata` is called outside of summary time, then this will incorrectly increment the summary number. And currently, there is nothing stopping this from happening. How about moving it to `addContainerStateToSummary`? You need a summary tree to call this method which means you should have summarized.

Reply:
Yes, I think `addContainerStateToSummary` is a better place if we expect to call this method at the end of summarizing. It's more explicit in terms of naming

Reply:
I also think `formMetadata` should be renamed to `getMetadataForSummary` or we can change it to `writeMetadataToSummary` to make it explicit that this is for adding to summary only.

Reply:
I'm always a fan of being more explicit with naming :)